### PR TITLE
use paginated github query

### PR DESCRIPTION
**Pull request by mashehu**

### CODE REVIEW

**Comment by mashehu** [comment]
@nf-core-bot fix linting

File: update_pipeline_details.php [hunk 1]
```diff
@@ -42,14 +43,37 @@ function get_gh_api($gh_api_url) {
...
-        die("\n-------- END ERROR " . date('Y-m-d h:i:s') . " --------\nCould not fetch $gh_api_url \n");
+    $first_page = true;
+    $next_page = false;
+    $res = [];
+    while ($first_page || $next_page) {
+        // reset loop vars
+        $first_page = false;
+        // Get GitHub API results
+        if ($next_page) {
+            $gh_api_url = $next_page;
```
Comment (ewels) [inline update_pipeline_details.php:54]: Where is this variable coming from? Just want to check that we're not breaking stuff by overwriting it here..

File: update_pipeline_details.php [hunk 2]
```diff
@@ -42,14 +43,37 @@ function get_gh_api($gh_api_url) {
...
+            die("\n-------- END ERROR " . date('Y-m-d h:i:s') . " --------\nCould not fetch $gh_api_url \n");
+        }
+
+        array_push($res, ...$tmp_results);
+        // Look for URL to next page of API results
+        $next_page = false;
+        $m_array = preg_grep('/rel="next"/', $http_response_header);
+
+        if (count($m_array) > 0) {
+            preg_match('/<([^>]+)>; rel="next"/', array_values($m_array)[0], $matches);
```
Comment (ewels) [inline update_pipeline_details.php:70]: Oof, this is the easiest way? There's no way to get the headers into an associative array or anything? Fair enough..

**Review by ewels — APPROVED** [review]
Two minor comments / queries. Otherwise LGTM ✅

File: update_pipeline_details.php (hunk 1 above)
Comment (mashehu) [reply update_pipeline_details.php:54]: it's the input parameter to this github query function

File: update_pipeline_details.php (hunk 2 above)
Comment (mashehu) [reply update_pipeline_details.php:70]: I could simply count the page up.

**Pull request merged**